### ISSUE TITLE
Missing info on failed deployment

### ISSUE THREAD

**Issue by jetzlstorfer** [keptn/keptn#2751]
TLDR:
When Keptn triggers a deployment with Helm that fails, e.g. due to insufficient CPUs on Kubernetes, there is currently no way to see this error in Keptn.

**To reproduce**:
1. Having a cluster with, e.g., 4vCPUs. 
1. Running a full tour tutorial with a 3 stage pipeline and sockshop as the project (cartsdb + carts).
1. Deploy cartsdb + carts with Keptn.
1. At the latest, when Keptn will move the carts microservice to the `staging` environment, the cluster will run out of CPUs. 
1. You won't find any indication of this problem in the bridge or in the CLI.

**Desired solution**:
Have an indication in the Keptn's Bridge, that the deployment has not succeeded + the reason why it failed.
In addition, a guide on how to start a new deployment once more CPUs are available is needed (e.g., point the user to a `helm delete release ...` command) However, a new deployment via Keptn is needed, otherwise the tests won't be triggered.

**Affected Component**

* [ ] Docs
* [x] CLI
* [x] Kubernetes Integration
* [ ] Openshift Integration
* [ ] Helm
* [ ] Istio
* [x] Bridge
* [ ] Gatekeeper
* [ ] Datastore
* [ ] REST API
* [x] eventbroker / distributor
* [ ] jmeter

**Comment by christian-kreuzberger-dtx** [comment]
FYI: We have an entry for this (although not K8s resources related) in our troubleshooting guide https://keptn.sh/docs/0.7.x/troubleshooting/#error-upgrade-failed-timed-out-waiting-for-the-condition - can you verify that this is correct?

Anyway, with 0.8.x we can provide a status flag for deployment.finished events, which would indicate that a deployment failed.

@johannes-b Can we also provide an error message in the deployment.finished event in this scenario?

**Comment by johannes-b** [comment]
The keptn/spec for the new CloudEvents knows the `message` property that can hold this information (please see: https://github.com/keptn/spec/blob/e04f13f41bff56d040076df27aa9a53a28b6ed47/cloudevents.md#data). 

I would like to use this issue to validate that Keptn 0.8 works according to the spec when it comes to failure situations, e.g., helm-service could not deploy, jmeter-service failed to run a test, etc.

**Comment by johannes-b** [comment]
This issue should verify: 
- `status` - [succeeded, errored, unknown] > The service ran into an internal error. The sequence execution stops.
- `result` - [pass, fail, `object`] > The result of the executed task. 
- `message` - Shows a message for more details.
